installer/src/tplink.rs

```diff
@@ -0,0 +1,297 @@
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::time::Duration;
+
+use anyhow::{Context, Error};
+use axum::{
+    Router,
+    body::{Body, to_bytes},
+    extract::{Request, State},
+    http::uri::Uri,
+    response::{IntoResponse, Response},
+    routing::any,
+};
+use bytes::{Bytes, BytesMut};
+use hyper::StatusCode;
+use hyper_util::{client::legacy::connect::HttpConnector, rt::TokioExecutor};
+use serde::Deserialize;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::TcpStream;
+use tokio::time::{sleep, timeout};
+
+use crate::InstallTpLink;
+
+type HttpProxyClient = hyper_util::client::legacy::Client<HttpConnector, Body>;
+
+pub async fn main_tplink(args: InstallTpLink) -> Result<(), Error> {
+    let InstallTpLink {
+        skip_sdcard,
+        admin_ip,
+    } = args;
```

Comment:
a possibly nicer way of writing this:

```suggestion
pub async fn main_tplink(InstallTpLink { skip_sdcard, admin_ip }: InstallTpLink) -> Result<(), Error> {
```

i could go either way on it, though